diff --git a/libfdt/libfdt_env.h b/libfdt/libfdt_env.h
index 73b6d40..650e3db 100644
--- a/libfdt/libfdt_env.h
+++ b/libfdt/libfdt_env.h
@@ -27,13 +27,13 @@ typedef uint32_t FDT_BITWISE fdt32_t;
 typedef uint64_t FDT_BITWISE fdt64_t;
 
 #define EXTRACT_BYTE(x, n)	((unsigned long long)((uint8_t *)&x)[n])
-#define CPU_TO_FDT16(x) ((EXTRACT_BYTE(x, 0) << 8) | EXTRACT_BYTE(x, 1))
-#define CPU_TO_FDT32(x) ((EXTRACT_BYTE(x, 0) << 24) | (EXTRACT_BYTE(x, 1) << 16) | \
-			 (EXTRACT_BYTE(x, 2) << 8) | EXTRACT_BYTE(x, 3))
-#define CPU_TO_FDT64(x) ((EXTRACT_BYTE(x, 0) << 56) | (EXTRACT_BYTE(x, 1) << 48) | \
-			 (EXTRACT_BYTE(x, 2) << 40) | (EXTRACT_BYTE(x, 3) << 32) | \
-			 (EXTRACT_BYTE(x, 4) << 24) | (EXTRACT_BYTE(x, 5) << 16) | \
-			 (EXTRACT_BYTE(x, 6) << 8) | EXTRACT_BYTE(x, 7))
+#define CPU_TO_FDT16(x) ((EXTRACT_BYTE(x, 1) << 8) | EXTRACT_BYTE(x, 0))
+#define CPU_TO_FDT32(x) ((EXTRACT_BYTE(x, 3) << 24) | (EXTRACT_BYTE(x, 2) << 16) | \
+			 (EXTRACT_BYTE(x, 1) << 8) | EXTRACT_BYTE(x, 0))
+#define CPU_TO_FDT64(x) ((EXTRACT_BYTE(x, 7) << 56) | (EXTRACT_BYTE(x, 6) << 48) | \
+			 (EXTRACT_BYTE(x, 5) << 40) | (EXTRACT_BYTE(x, 4) << 32) | \
+			 (EXTRACT_BYTE(x, 3) << 24) | (EXTRACT_BYTE(x, 2) << 16) | \
+			 (EXTRACT_BYTE(x, 1) << 8) | EXTRACT_BYTE(x, 7))
 
 static inline uint16_t fdt16_to_cpu(fdt16_t x)
 {
